R/outcome_multi.R

```diff
@@ -5,53 +5,36 @@
 
 #' Use gsynth to fit factor model with 
 #' @importFrom utils capture.output
+#' @param long_df A long dataframe with 4 columns in the order unit, time, trt, outcome
 #' @param X Matrix of outcomes
 #' @param trt Vector of treatment status for each unit
 #' @param r Number of factors to use (or start with if CV==1)
 #' @param r.end Max number of factors to consider if CV==1
 #' @param force Fixed effects (0=none, 1=unit, 2=time, 3=two-way)
 #' @param CV Whether to do CV (0=no CV, 1=yes CV)
-#'
+#' @noRd
 #' @return \itemize{
 #'           \item{y0hat }{Predicted outcome under control}
 #'           \item{params }{Regression parameters}}
-fit_gsynth_multi <- function(X, trt, r=0, r.end=5, force=3, CV=1) {
-
+fit_gsynth_multi <- function(long_df, X, trt, r=0, force=3, CV=1) {
     if(!requireNamespace("gsynth", quietly = TRUE)) {
         stop("In order to fit generalized synthetic controls, you must install the gsynth package.")
     }
-
-
     ttot <- ncol(X)
     n <- nrow(X)
-    
-    ## observed matrix (everything observed)
-    I <- matrix(1, ttot, n)
-
-    ## treatment matrix
-    trt_mat <- matrix(0, nrow=n, ncol=ttot)
-    trt_mat[is.finite(trt),] <-
-        t(vapply(trt[is.finite(trt)],
-               function(ti) c(rep(0, ti), rep(1, ttot-ti)),
-               numeric(ttot)))
 
-    ## use internal gsynth function
-    capture.output(gsyn <- gsynth:::synth.core(t(X), NULL, t(trt_mat), I,
-                                               r=r, r.end=r.end,
-                                               force=force, CV=CV,
-                                               tol=0.001))
-    ## get predicted outcomes
+    labels <- colnames(long_df)
+    gsyn <- gsynth::gsynth(data = long_df, Y = labels[4], D = labels[3], index = c(labels[1], labels[2]), force = force, CV = CV, r=r)
```

Comment:
We had talked about adding a more understandable warning if `r` is set too high. Not sure we need to do that at the moment, but it's something to keep in mind

Reply:
That's super weird; i remember implementing a try catch for this. I'll re-implement one here.

Reply:
Oh wait, I did implement this. The try catch is upstream in `multisynth_class.R`.